app/bundles/LeadBundle/Model/LeadModel.php

```diff
@@ -1958,6 +1960,113 @@ public function setUtmTags(Lead $lead, UtmTag $utmTags)
     }
 
     /**
+     * Add leads UTM tags via API.
+     *
+     * @param Lead  $lead
+     * @param array $query
+     */
+    public function addUTMTags(Lead $lead, $query)
+    {
+        // available fields and it's setter.
+        $fields = ['utm_campaign' => 'UtmCampaign',
+                   'utm_source'   => 'UtmSource',
+                   'utm_medium'   => 'UtmMedium',
+                   'utm_content'  => 'UtmContent',
+                   'utm_term'     => 'UtmTerm',
+                   'useragent'    => 'UserAgent',
+                   'url'          => 'Url',
+                   'referer'      => 'Referer',
+                   'query'        => 'Query',
+                   'remotehost'   => 'RemoteHost',
+                   'lastActive'   => 'DateAdded',
+        ];
+
+        // 'query' must be an array
+        if (isset($query['query']) && !is_array($query['query'])) {
+            parse_str($query['query'], $queryResult);
+            if (!empty($queryResult)) {
+                $query['query'] = $queryResult;
+            } else {
+                // Something wrong with it
+                unset($query['query']);
+            }
+        }
+
+        // see if active date set, so we can use it
+        $updateLastActive = false;
+        $lastActive       = new \DateTime();
+        // should be: yyyy-mm-ddT00:00:00+00:00
+        if (isset($query['lastActive'])) {
+            $lastActive       = new \DateTime($query['lastActive']);
+            $updateLastActive = true;
+        }
+        $query['lastActive'] = $lastActive;
+
+        // New utmTag
+        $utmTags = new UtmTag();
+
+        // cycle through calling appropriate setter
+        foreach ($fields as $q => $prop) {
+            if (isset($query[$q])) {
+                $setter = 'set'.$prop;
+                $utmTags->$setter($query[$q]);
+            }
+        }
```

Comment:
I had hoped to use a generic routine (change Studly to Camel case) but there are fields that don't follow the standard so I don't think it will work.  ('Useagent -> UserAgent', 'remotehost' -> 'remoteHost')